cloudflare.py

```diff
@@ -242,14 +242,21 @@ class Zone(object):
 
     def __init__(self, name, zone):
         self.name = name
-        self.auth_email = zone["auth_email"]
-        self.auth_key = zone["auth_key"]
+        self.api_token = zone.get("api_token", None)
+        self.auth_email = zone.get("auth_email", None)
+        self.auth_key = zone.get("auth_key", None)
         self.zone_id = zone["zone_id"]
         self.records = zone["records"]
         self.exclude = zone.get('exclude', [])
+        
+        if not self.api_token and (not self.auth_email or not self.auth_key):
```

Comment:
Just a nitpick: 
To me this would read better as 

```
if not self.api_token and not (self.auth_email and self.auth_key):
```

But that might just be personal preference

Reply:
As you wish. I think it is more straight forward to understand as well.